apps/builder/app/builder/features/workspace/canvas-tools/resize-handles.tsx

```diff
@@ -78,59 +77,63 @@ const handleIcon = (
   </svg>
 );
 
-const baseDragProps = {
-  startDistanceThreashold: 0,
-  elementToData: () => true,
-  onStart() {
-    isCanvasPointerEventsEnabledStore.set(false);
-    isResizingCanvasStore.set(true);
-  },
-  onEnd() {
-    isCanvasPointerEventsEnabledStore.set(true);
-    isResizingCanvasStore.set(false);
-  },
+const updateBreakpoint = (width: number) => {
+  const applicableBreakpoint = findApplicableMedia(
+    Array.from(breakpointsStore.get().values()),
+    width
+  );
+  if (applicableBreakpoint) {
+    selectedBreakpointIdStore.set(applicableBreakpoint.id);
+  }
 };
 
-const useResize = () => {
-  const isDragging = useStore(isCanvasPointerEventsEnabledStore) === false;
-  const [isHovering, setIsHovering] = useState(false);
+const useScrub = ({ side }: { side: "right" | "left" }) => {
+  const ref = useRef<HTMLDivElement>(null);
 
-  const updateBreakpoint = (width: number) => {
-    const applicableBreakpoint = findApplicableMedia(
-      Array.from(breakpointsStore.get().values()),
-      width
-    );
-    if (applicableBreakpoint) {
-      selectedBreakpointIdStore.set(applicableBreakpoint.id);
+  useEffect(() => {
+    if (ref.current === null) {
+      return;
     }
-  };
+    const scrub = numericScrubControl(ref.current, {
+      getInitialValue() {
+        return canvasWidthStore.get();
+      },
+      getValue(state, movement) {
+        const value =
+          side === "left"
+            ? state.value - movement * 2
```

Comment:
What is 2? Speed of movement? Probably better make it const with some good name and explain why 2 is selected. Also does this mean that all values are odd or even depending on initial value?

Reply:
will add a comment, yes, it's compensating because we center the canvas, so that cursor stays at the edge